app/isbn.hs

```diff
@@ -0,0 +1,48 @@
+import Data.List
```

Comment:
simplify your logic for this solution. Think of a better way to deal with the presence of `-`